float32 speed
---
bool result